Source/AliveLibAO/SaveGame.cpp

```diff
@@ -1,13 +1,306 @@
+};
+
+const unsigned char byte_4BC590[] =
+{
+    0x82, 0x72, 0x82, 0x83, 0x82, 0x92, 0x82, 0x81, 0x82, 0x82,
+    0x82, 0x81, 0x82, 0x8E, 0x82, 0x89, 0x82, 0x81, 0x82, 0x8E,
+    0x81, 0x40, 0x82, 0x6D, 0x82, 0x85, 0x82, 0x93, 0x82, 0x94,
+    0x82, 0x93, 0x81, 0x40, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+};
+
+const unsigned char byte_4BC5B8[] =
+{
+    0x82, 0x72, 0x82, 0x94, 0x82, 0x8F, 0x82, 0x83, 0x82, 0x8B,
+    0x82, 0x99, 0x82, 0x81, 0x82, 0x92, 0x82, 0x84, 0x81, 0x40,
+    0x82, 0x64, 0x82, 0x93, 0x82, 0x83, 0x82, 0x81, 0x82, 0x90,
+    0x82, 0x85, 0x81, 0x40, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+};
+
+const unsigned char byte_4BC5E0[] =
+{
+    0x82, 0x72, 0x82, 0x94, 0x82, 0x8F, 0x82, 0x83, 0x82, 0x8B,
+    0x82, 0x99, 0x82, 0x81, 0x82, 0x92, 0x82, 0x84, 0x81, 0x40,
+    0x82, 0x64, 0x82, 0x93, 0x82, 0x83, 0x82, 0x81, 0x82, 0x90,
+    0x82, 0x85, 0x81, 0x40, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+};
+
+const BYTE byte_4BC608[] =
+{
+    0x82, 0x72, 0x82, 0x94, 0x82, 0x8F, 0x82, 0x83, 0x82, 0x8B,
+    0x82, 0x99, 0x82, 0x81, 0x82, 0x92, 0x82, 0x84, 0x81, 0x40,
+    0x82, 0x71, 0x82, 0x85, 0x82, 0x94, 0x82, 0x95, 0x82, 0x92,
+    0x82, 0x8E, 0x81, 0x40, 0x00, 0x00
+};
+
+using byteArray = decltype(byte_4BC450);
+const unsigned char* const dword_4BC62C[] =
+{
+    nullptr,
+    byte_4BC450,
+    byte_4BC5B8,
+    byte_4BC4C8,
+    byte_4BC4F0,
+    byte_4BC5E0,
+    byte_4BC608,
+    nullptr,
+    byte_4BC540,
+    byte_4BC568,
+    nullptr,
+    nullptr,
+    byte_4BC4A0
+};
+
+ALIVE_VAR(1, 0x4CF2B0, Save_PSX_Header, sSaveHeader2_4CF2B0, {});
+ALIVE_VAR(1, 0x4BC250, Save_PSX_Header, sSaveHeader1_4BC250, {});
+ALIVE_VAR(1, 0x5076B4, WORD, bUseAltSaveHeader_5076B4, 0);
+
+void CC SaveGame::Save_459490(SaveData* pSaveData)
+{
+    Save_PSX_Header* pHeaderToUse = nullptr;
+    if (bUseAltSaveHeader_5076B4)
+    {
+        pHeaderToUse = &sSaveHeader1_4BC250;
+    }
+    else
+    {
+        pHeaderToUse = &sSaveHeader2_4CF2B0;
+    }
+
+    pSaveData->field_0_header = *pHeaderToUse;
+
+    //TODO NUKE CASTS
+    sub_45A2D0(
+        (BYTE *)&pSaveData->field_0_header.field_0_frame_1_name[4],
+        (BYTE *) &dword_4BC62C[static_cast<int>(gMap_507BA8.field_0_current_level)],
+        34
+    );
+
+    char path = 0;
+    if (gMap_507BA8.field_0_current_level == LevelIds::eRuptureFarmsReturn_13)
+    {
+        short path_id = -1;
+        for (short i = 0; i < 4; i++)
+        {
+            for(int j = 0; j < 8; j++ )
+            {
+                path = word_4BC670[i][j];
+                if (!path)
+                {
+                    break;
+                }
+                if (gMap_507BA8.field_2_current_path == path)
+                {
+                    path_id = i;
+                    i = 4;
+                    break;
+                }
+            }
+        }
+        if (path_id != -1)
+        {
+            pSaveData->field_0_header.field_0_frame_1_name[44] = -127;
```

Comment:
I think these are unsigned, change to hex in IDA to figure out the values, prob 0x8X